### PR TITLE
feat: support responseStatusInterceptor

### PR DESCRIPTION
**Pull request by gaoting**

### CODE REVIEW

File: src/interceptors/responseStatusInterceptor.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+export interface ResponseStatusInterceptorOptions {
+  handler?: Record<number, (response: any) => Record<string, string>> | any
+  axiosInterceptorOptions?: AxiosInterceptorOptions
+}
+
+export function responseStatusInterceptor(options: ResponseStatusInterceptorOptions): ResponseInterceptor {
+  const proxyResponse = (newResponse: any, type: string): any => {
+    const { handler } = options
+    const keys = Object.keys(handler)
+    const status = type === 'success' ? newResponse.status : newResponse.response.status
```
Comment (zhangmo8) [inline src/interceptors/responseStatusInterceptor.ts:15]: `newResponse` rename to `originalResponse`

File: src/interceptors/responseStatusInterceptor.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import type { AxiosInterceptorOptions, AxiosResponse } from 'axios'
+import type { ResponseInterceptor } from '../instance'
+import { createMatcher } from '../matcher'
+import { isFunction } from '@varlet/shared'
+
+export interface ResponseStatusInterceptorOptions {
+  handler?: Record<number, (response: any) => Record<string, string>> | any
+  axiosInterceptorOptions?: AxiosInterceptorOptions
```
Comment (haoziqaq) [inline src/interceptors/responseStatusInterceptor.ts:8]: need support include and exclude

File: src/interceptors/responseStatusInterceptor.ts (hunk 1 above)
Comment (haoziqaq) [inline src/interceptors/responseStatusInterceptor.ts:15]: error and response are two different types and need to be handled differently

**Comment by haoziqaq** [comment]
Please complete `src/interceptors/responseStatusInterceptor.md`

File: src/interceptors/responseStatusInterceptor.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      if (!matcher(response.config.method ?? '', response.config.url ?? '')) {
+        return response
+      }
+
+      const keys = Object.keys(handlerCode)
+      const key = keys.find((currentKey) => currentKey === response.status.toString())
+      if (!handlerCode[key as unknown as number] && !isFunction(handlerCode[key as unknown as number])) {
+        throw Error(`[Axle] ${key} handler should be a function`)
+      }
+      handlerCode[key as unknown as number](response)
```
Comment (zhangmo8) [inline src/interceptors/responseStatusInterceptor.ts:29]: should be return the result after processing by handler

File: src/interceptors/responseStatusInterceptor.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        return response
+      }
+
+      const keys = Object.keys(handlerCode)
+      const key = keys.find((currentKey) => currentKey === response.status.toString())
+      if (!handlerCode[key as unknown as number] && !isFunction(handlerCode[key as unknown as number])) {
+        throw Error(`[Axle] ${key} handler should be a function`)
+      }
+      const responseData = handlerCode[key as unknown as number](response)
+      return responseData || response
```
Comment (zhangmo8) [inline src/interceptors/responseStatusInterceptor.ts:30]: `||` replace to `??`

File: README.zh-CN.md
Comment (haoziqaq) [inline README.zh-CN.md:1]: delete this file

File: playground/README.md
Comment (haoziqaq) [inline playground/README.md:1]: delete this file

File: src/interceptors/responseStatusInterceptor.ts (hunk 4 above)
Comment (zhangmo8) [inline src/interceptors/responseStatusInterceptor.ts:30]:
```suggestion
      return responseData ?? response
```

File: playground/src/App.vue [hunk 5]
```diff
@@ -81,6 +82,15 @@ const [errorUser, throwError, { loading: isThrowErrorLoading }] = useThrowError(
...
 
+const [_, proxyError, { loading: isProxyEroorLoading }] = useProxyError({
+  onBefore() {
+    Snackbar.loading('Proxy!')
+  },
+  onError(error) {
+    Snackbar.error(error.toString())
+  },
+})
+
```
Comment (haoziqaq) [inline playground/src/App.vue:93]: delete the test code related to this interceptor

File: src/interceptors/examples/responseStatusInterceptor.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
+# Response Status Interceptor
+
+```js
+import { createAxle, responseStatusInterceptor } from '@varlet/axle'
+
+const axle = createAxle()
+
+axle.useResponseInterceptor(
+    responseStatusInterceptor({
```
Comment (haoziqaq) [inline src/interceptors/examples/responseStatusInterceptor.md:9]: format this file

File: src/interceptors/responseStatusInterceptor.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import type { AxiosInterceptorOptions, AxiosResponse } from 'axios'
+import type { ResponseInterceptor } from '../instance'
+import { createMatcher } from '../matcher'
+import { isFunction } from '@varlet/shared'
+
+export interface ResponseStatusInterceptorOptions {
+  handlerCode?: Record<number, (response: any) => Record<string, string>> | any
+  handlerError?: Record<number, (response: any) => Record<string, string>> | any
```
Comment (haoziqaq) [inline src/interceptors/responseStatusInterceptor.ts:8]:
```js
export interface ResponseStatusInterceptorOptions {
  validateHandler?: Record<number, (response: AxiosResponse<any, any>) => any
  invalidateHandler?: Record<number, (response: AxiosResponse<any, any>) => any
  ...
}
```

File: src/interceptors/responseStatusInterceptor.ts [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      const matcher = createMatcher(options.include, options.exclude)
+      if (!matcher(response.config.method ?? '', response.config.url ?? '')) {
+        return response
+      }
+
+      const keys = Object.keys(handlerCode)
+      const key = keys.find((currentKey) => currentKey === response.status.toString())
+      if (!handlerCode[key as unknown as number] && !isFunction(handlerCode[key as unknown as number])) {
+        throw Error(`[Axle] ${key} handler should be a function`)
+      }
```
Comment (haoziqaq) [inline src/interceptors/responseStatusInterceptor.ts:28]: remove this

File: src/interceptors/responseStatusInterceptor.ts (hunk 4 above)
Comment (haoziqaq) [inline src/interceptors/responseStatusInterceptor.ts:30]: responseData == null ? response : responseData

File: src/interceptors/responseStatusInterceptor.ts [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+      const keys = Object.keys(handlerCode)
+      const key = keys.find((currentKey) => currentKey === error.response.status.toString())
+      if (!handlerCode[key as unknown as number] && !isFunction(handlerCode[key as unknown as number])) {
+        throw Error(`[Axle] ${key} handler should be a function`)
+      }
+      const errorData = handlerCode[key as unknown as number](error)
+      return errorData || error
+    },
+    options: options.axiosInterceptorOptions,
+  }
```
Comment (haoziqaq) [inline src/interceptors/responseStatusInterceptor.ts:50]: logic error. please deal with the invalidate situation directly.

File: src/interceptors/responseStatusInterceptor.ts (hunk 4 above)
Comment (zhangmo8) [reply src/interceptors/responseStatusInterceptor.ts:30]: This content already be suggestion in mine review

**Pull request merged**